pkg/routeagent_driver/handlers/calico/ippool_handler.go

```diff
@@ -0,0 +1,198 @@
+		return nil
+	}
+
+	err := h.deleteIPPool(endpoint)
+
+	return errors.Wrap(err, "failed to handle RemoteEndpointRemoved event")
+}
+
+func (h *calicoIPPoolHandler) TransitionToNonGateway() error {
+	logger.Info("TransitionToNonGateway")
+
+	h.isGateway = false
+
+	return nil
+}
+
+func (h *calicoIPPoolHandler) TransitionToGateway() error {
+	logger.Info("TransitionToGateway")
+
+	retErr := error(nil)
+	h.isGateway = true
+
+	for _, endpoint := range h.remoteEndpoints {
+		err := h.createIPPool(endpoint)
+		if err != nil {
+			logger.Warningf("Failed to create ippool %s", endpoint.GetName())
+
+			retErr = err
+		}
+	}
+
+	return retErr
+}
+
+func (h *calicoIPPoolHandler) Stop(uninstall bool) error {
+	if !uninstall || !h.isGateway {
+		return nil
+	}
+
+	logger.Info("Uninstalling Calico IPPools used for Submariner")
+
+	labelSelector := labels.SelectorFromSet(map[string]string{submarinerIPPool: "true"}).String()
+	err := h.client.ProjectcalicoV3().IPPools().DeleteCollection(context.TODO(), metav1.DeleteOptions{},
+		metav1.ListOptions{LabelSelector: labelSelector})
+
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errors.Wrapf(err, "Failed to delete Calico IPPools using labelSelector %q", labelSelector)
+	}
+
+	logger.Infof("Successfully delete Calico IPPools using labelSelector %q", labelSelector)
+
+	return nil
+}
+
+func (h *calicoIPPoolHandler) createIPPool(endpoint *submV1.Endpoint) error {
+	subnets := cidr.ExtractIPv4Subnets(endpoint.Spec.Subnets)
+	retErr := error(nil)
+
+	for _, subnet := range subnets {
+		iPPoolObj := &calicoapi.IPPool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   getEndpointSubnetIPPoolName(endpoint, subnet),
+				Labels: map[string]string{submarinerIPPool: "true"},
+			},
+			Spec: calicoapi.IPPoolSpec{
+				CIDR:        subnet,
+				NATOutgoing: false,
+				Disabled:    true,
+			},
+		}
+
+		if _, err := h.client.ProjectcalicoV3().IPPools().Create(context.TODO(), iPPoolObj,
+			metav1.CreateOptions{}); err != nil && !apierrors.IsAlreadyExists(err) {
+			logger.Errorf(err, "Error creating Calico IPPool for ClusterID %q subnet %q", endpoint.Spec.ClusterID, subnet)
+			retErr = err
+		} else {
+			logger.Infof("Successfully create Calico IPPool %q", iPPoolObj.GetName())
+		}
+	}
+
+	return retErr
+}
+
+func (h *calicoIPPoolHandler) deleteIPPool(endpoint *submV1.Endpoint) error {
+	retErr := error(nil)
+	subnets := cidr.ExtractIPv4Subnets(endpoint.Spec.Subnets)
+
+	for _, subnet := range subnets {
+		poolName := getEndpointSubnetIPPoolName(endpoint, subnet)
+		if err := h.client.ProjectcalicoV3().IPPools().Delete(context.TODO(),
+			poolName, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+			logger.Errorf(err, "Error deleting Calico IPPool for ClusterID %q subnet %q", endpoint.Spec.ClusterID, subnet)
+			retErr = err
+		} else {
+			logger.Infof("Successfully delete Calico IPPool %q", poolName)
+		}
+	}
```

Comment:
We shouldn't log success if not found:

```suggestion
		err := h.client.ProjectcalicoV3().IPPools().Delete(context.TODO(),
			poolName, metav1.DeleteOptions{})
		if err == nil {
		    logger.Infof("Successfully deleted Calico IPPool %q", poolName)
		    continue
		}
		    
		if !apierrors.IsNotFound(err) {
		    retErrors = append(retErrors, errors.Wrapf(err, "error deleting Calico IPPool for ClusterID %q subnet %q", endpoint.Spec.ClusterID, subnet)))
	}
```